File: 프로그래머스/Level 1/둘만의 암호.swift
// https://school.programmers.co.kr/learn/courses/30/lessons/155652
import Foundation

func solution(_ s:String, _ skip:String, _ index:Int) -> String {
    var answer: String = ""
    var s = s.map { String($0) }
    var skip = skip.map { String($0) }
    var alp = "abcdefghijklmnopqrstuvwxyz".map { String($0) }
    
    for i in 0..<s.count {
        var idx = alp.firstIndex(of: s[i])!
        var cnt = 0
        
        while cnt < index {
            idx += 1
            if idx >= 26 {
                idx = 0;
            }
            
            if skip.contains(alp[idx]) {
                continue
            }
            
            cnt += 1
        }
        
        answer += alp[idx]
    }
    return answer
}

File: 프로그래머스/Level 1/둘만의 암호.cpp
// https://school.programmers.co.kr/learn/courses/30/lessons/155652
#include <string>
#include <vector>

using namespace std;

int contains(string str, char ch) {
    for (char c: str) {
        if (c == ch) return 1;
    }
    return 0;
}

string solution(string s, string skip, int index) {
    string answer = "";
    
    for (char ch: s) {
        int val = ch - 'a';
        int cnt = 0;
        
        while (cnt < index) {
            val++;
            if (val > 25) val = 0;
            
            if (contains(skip, val+'a')) continue;
            cnt++;
        }
        answer.push_back(val + 'a');
    }
    
    return answer;
}